pandas/core/frame.py

```diff
@@ -6068,19 +6068,32 @@ def _gotitem(self,
     def aggregate(self, func, axis=0, *args, **kwargs):
         axis = self._get_axis_number(axis)
 
-        # TODO: flipped axis
         result = None
-        if axis == 0:
-            try:
-                result, how = self._aggregate(func, axis=0, *args, **kwargs)
-            except TypeError:
-                pass
+        try:
+            result, how = self._aggregate(func, axis=axis, *args, **kwargs)
+        except TypeError:
+            pass
         if result is None:
             return self.apply(func, axis=axis, args=args, **kwargs)
         return result
 
+    def _aggregate(self, arg, axis=0, *args, **kwargs):
+        axis = self._get_axis_number(axis)
```

Comment:
can you add a comment here on what you are doing

Reply:
This probably isn't necessary here, as _aggregate is an internal function and ``axis = self._get_axis_number(axis)`` is already called in ``aggregate´´. I'll just delete that line an dexplain the rest.